e2e/browser/test-app/components/accessGrants/index.tsx

```diff
@@ -90,34 +93,42 @@ export default function AccessGrant({
     setSharedResourceIri(undefined);
   };
 
-  const handleGrant = async () => {
+  const handleRequest = async () => {
     if (typeof sharedResourceIri !== "string") {
       // If the resource does not exist, do nothing.
       return;
     }
-    const accessGrantRequest = await approveAccessRequest(
-      undefined,
-      {
-        requestor: "https://johndoe.webid",
-        access: { read: true },
-        resources: [sharedResourceIri],
-      },
-      {
-        fetch: session.fetch,
-      },
-    );
-    setAccessGrant(accessGrantRequest);
+    // if the resource owner is not defined, do nothing.
+    if (session.info.webId) {
```

Comment:
This test could be added to the clause of line 97. It's usually more readable to have an early return (and so to test for negative conditions) rather than testing for positive conditions and having the "main" code inside the conditional branch.